### PR TITLE
noetic-devel

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Merge after https://github.com/tue-robotics/hmi/pull/25
Merge together with https://github.com/tue-robotics/action_server/pull/143

### CODE REVIEW

File: challenge_demo/CMakeLists.txt [hunk 1]
```diff
@@ -28,11 +28,11 @@ catkin_python_setup()
 ## * Let MSG_DEP_SET be the set of packages whose message types you use in
 ##   your messages/services/actions (e.g. std_msgs, actionlib_msgs, ...).
 ## * In the file package.xml:
-##   * add a build_depend and a run_depend tag for each package in MSG_DEP_SET
+##   * add a build_depend and a exec_depend tag for each package in MSG_DEP_SET
```
Comment (jlunenburg) [inline challenge_demo/CMakeLists.txt:31]: Remove this commented stuff entirely?

File: challenge_eegpsr/CMakeLists.txt [hunk 2]
```diff
@@ -28,11 +28,11 @@ find_package(catkin REQUIRED COMPONENTS
 ## * Let MSG_DEP_SET be the set of packages whose message types you use in
 ##   your messages/services/actions (e.g. std_msgs, actionlib_msgs, ...).
 ## * In the file package.xml:
-##   * add a build_depend and a run_depend tag for each package in MSG_DEP_SET
+##   * add a build_depend and a exec_depend tag for each package in MSG_DEP_SET
```
Comment (jlunenburg) [inline challenge_eegpsr/CMakeLists.txt:31]: Same here, remove the commented stuff

File: challenge_eegpsr/src/test_recognition.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 # ----------------------------------------------------------------------------------------------------
 
 def usage():
-    print """Usage:
+    print("""Usage:
```
Comment (jlunenburg) [inline challenge_eegpsr/src/test_recognition.py:16]: Add import? In any way, please be consistent

File: challenge_eegpsr/test/navigate_in_front_of.py [hunk 4]
```diff
@@ -15,7 +15,7 @@
         ids = [e.id for e in robot.ed.get_entities() if e.is_a("furniture")]
         random.shuffle(ids)
 
-    print "IDS:", ids
+    print("IDS:", ids)
```
Comment (jlunenburg) [inline challenge_eegpsr/test/navigate_in_front_of.py:18]: Idem?

File: challenge_final/src/challenge_final/get_orders.py [hunk 5]
```diff
@@ -1,19 +1,11 @@
...
-import cv2
-import numpy as np
-from datetime import datetime
-
 # ROS
 import rospy
 import smach
-import cv_bridge
-import rospkg
+
```
Comment (jlunenburg) [inline challenge_final/src/challenge_final/get_orders.py:4]: I didn't check all these imports

File: robocup_knowledge/src/robocup_knowledge/environments/rgo2018/common.py [hunk 6]
```diff
@@ -216,40 +215,40 @@ def get_object_color(obj):
 
 # Returns (location, area_name)
 def get_object_category_location(obj_cat):
-    location = category_locations[obj_cat].keys()[0]
-    area_name = category_locations[obj_cat].values()[0]
-    return (location, area_name)
+    location = next(iter(category_locations[obj_cat].keys()))[0]
+    area_name = next(iter(category_locations[obj_cat].values()))[0]
```
Comment (jlunenburg) [inline robocup_knowledge/src/robocup_knowledge/environments/rgo2018/common.py:219]: I don't think you need the indexing in these two lines.

File: robocup_knowledge/src/robocup_knowledge/environments/rgo2019/common.py [hunk 7]
```diff
@@ -266,37 +265,37 @@ def get_object_weight(obj):
 
 # Returns (location, area_name)
 def get_object_category_location(obj_cat):
-    location = category_locations[obj_cat].keys()[0]
-    area_name = category_locations[obj_cat].values()[0]
-    return (location, area_name)
+    location = next(iter(category_locations[obj_cat].keys()))[0]
```
Comment (jlunenburg) [inline robocup_knowledge/src/robocup_knowledge/environments/rgo2019/common.py:268]: Indexing

File: robocup_knowledge/src/robocup_knowledge/environments/rwc2016_common/common.py [hunk 8]
```diff
@@ -105,40 +106,38 @@ def get_object_category(obj):
 
 # Returns (location, area_name)
 def get_object_category_location(obj_cat):
-    if not obj_cat in category_locations:
-        return None
-    else:
-        (location, area_name) = category_locations[obj_cat]
-        return (location, area_name)
+    location = next(iter(category_locations[obj_cat].keys()))[0]
```
Comment (jlunenburg) [inline robocup_knowledge/src/robocup_knowledge/environments/rwc2016_common/common.py:109]: Indexing

File: robocup_knowledge/src/robocup_knowledge/environments/rwc2017/common.py [hunk 9]
```diff
@@ -216,36 +216,37 @@ def get_object_category(obj):
 
 def get_object_category_location(obj_cat):
     # Returns (location, area_name)
-    location = category_locations[obj_cat].keys()[0]
-    area_name = category_locations[obj_cat].values()[0]
-    return (location, area_name)
+    location = next(iter(category_locations[obj_cat].keys()))[0]
```
Comment (jlunenburg) [inline robocup_knowledge/src/robocup_knowledge/environments/rwc2017/common.py:219]: Indexing

File: robocup_knowledge/src/robocup_knowledge/environments/rwc2018/common.py [hunk 10]
```diff
@@ -233,37 +232,37 @@ def get_object_weight(obj):
 
 # Returns (location, area_name)
 def get_object_category_location(obj_cat):
-    location = category_locations[obj_cat].keys()[0]
-    area_name = category_locations[obj_cat].values()[0]
-    return (location, area_name)
+    location = next(iter(category_locations[obj_cat].keys()))[0]
```
Comment (jlunenburg) [inline robocup_knowledge/src/robocup_knowledge/environments/rwc2018/common.py:235]: Indexing

File: robocup_knowledge/src/robocup_knowledge/environments/siza_demo/common.py [hunk 11]
```diff
@@ -173,39 +172,40 @@ def get_object_color(obj):
 
 # Returns (location, area_name)
 def get_object_category_location(obj_cat):
-    location = category_locations[obj_cat].keys()[0]
-    area_name = category_locations[obj_cat].values()[0]
-    return (location, area_name)
+    location = next(iter(category_locations[obj_cat].keys()))[0]
```
Comment (jlunenburg) [inline robocup_knowledge/src/robocup_knowledge/environments/siza_demo/common.py:175]: Indexing

**Review by jlunenburg — COMMENTED** [review]
This is a lot to review ;-). Skimming through it, I noticed a couple of things that might be improved:
* I had the feeling that the way you updated the 'knowledge' doesn't work (didn't test it though)
* importing `__future__` didn't seem to be done consistently (either do it everywhere or don't do it)
* you basically added everything to every `__init__.py`. Without checking it thoroughly, does this mean everything is available at the top level? That would't be convenient

File: robocup_knowledge/src/robocup_knowledge/environments/rgo2019/common.py (hunk 7 above)
Comment (MatthijsBurgh) [reply robocup_knowledge/src/robocup_knowledge/environments/rgo2019/common.py:268]: What do you mean?

**Review by jlunenburg — APPROVED** [review]
I gave my feedback once, now I put my faith in the lap of CI ;-)

**Pull request merged**